libraries/conf_helpers.rb

```diff
@@ -27,35 +39,74 @@ def self.read_config(conf_path)
     end
 
     def self.parse_config(conf_body)
-      current_stanza = 'default'
+      current_section = 'default'
       current_config = {}
 
       conf_body.each_line do |ln|
         case ln
         when /^\s*\[([^#]+)\]\s*/
-          current_stanza = Regexp.last_match[1]
+          current_section = Regexp.last_match[1]
         when /^\s*([^#]+?)\s*=\s*([^#]+?)\s*(?:#.*)?$/
-          (current_config[current_stanza] ||= {})[Regexp.last_match[1]] = Regexp.last_match[2]
+          (current_config[current_section] ||= {})[Regexp.last_match[1]] = Regexp.last_match[2]
         end
       end
 
       current_config
     end
 
     def self.merge_config(current_config, desired_config)
```

Comment:
1. merge_config isn't pulling data from disk, all config is passed in the two hash parameters.
2. Moving nil filtering to evaluate makes less sense. It does handle both current and desired config, but it only passes current config to the procs. merge_config deals with combining the current and desired configs and seems to be the most appropriate place for this. I could extract it into a method, but I don't see any value in that.
3. Using a template is a good idea... I can move the logic of building the conf file to the resource.